app/components/CrashRecommendationCard.tsx

```diff
@@ -0,0 +1,294 @@
+import { useState } from "react";
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Form from "react-bootstrap/Form";
+import Spinner from "react-bootstrap/Spinner";
+import { useForm } from "react-hook-form";
+import { useQuery, useMutation } from "@/utils/graphql";
+import CrashRecommendationParters from "./CrashRecommendationPartners";
+import {
+  RECOMMENDATION_STATUS_QUERY,
+  RECOMMENDATION_PARTNERS_QUERY,
+  INSERT_RECOMMENDATION_MUTATION,
+  UPDATE_RECOMMENDATION_MUTATION,
+} from "@/queries/recommendations";
+import {
+  Recommendation,
+  RecommendationStatus,
+  RecommendationPartner,
+  CoordinationPartner,
+  RecommendationFormInputs,
+} from "@/types/recommendation";
+import { useAuth0 } from "@auth0/auth0-react";
+
+/**
+ * Compares the old vs new RecommendationPartner arrays and returns
+ * new arrays of partner data that can be used in graphql mutation
+ * @param recommendationId - the pk of the current recommendation, if we are editing
+ * @param partnersOld - array of RecommendationPartner[] objs
+ * @param partnersNew - array of RecommendationPartner's from the form data. it is
+ * expected that these objects may only have a `partner_id` property
+ * @returns [addPartners, deletePartnerPks]
+ */
+const getPartnerChanges = (
```

Comment:
Should move this to a file in `/utils`...?

Reply:
I'm torn between moving it out of the component to reduce the file line count and not moving it because it probably won't be reused anywhere. I'm leaning towards moving to utils because I've been liking moving custom hook into a `hooks` folder to keep components tidier.

Reply:
i think i lean towards leaving this function here because like mike said it wont be reused elsewhere and i dont think its ridiculously long? if we move this it starts feeling like where do we draw the line on moving functions to the utils folders and are we just making things more complicated and hard to follow at that point?

Reply:
yeah. i agree that potential future reusability is a good test of when to move into the general `utils` folder. and i am coming around on the idea of every component living in a directory with a standard pattern for local utils, hooks, const files...